src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,60 @@
+package duke.parser;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class Parser {
+
+    public String parseCommand(String s) {
+        if (s.toLowerCase().equals("bye")) {
+            return "bye";
+        } else if (s.toLowerCase().equals("list")) {
+            return "list";
+        } else if (s.toLowerCase().startsWith("todo")) {
+            return "todo";
+        } else if (s.toLowerCase().startsWith("done")) {
+            return "done";
+        } else if (s.toLowerCase().startsWith("find")) {
+            return "find";
+        } else if (s.toLowerCase().startsWith("event")) {
+            return "event";
+        } else if (s.toLowerCase().startsWith("delete")) {
+            return "delete";
+        } else if (s.toLowerCase().startsWith("deadline")) {
+            return "deadline";
+        } else {
+            return "unknown command detected";
+        }
+    }
+
+    public String parseDesc(String s) {
+        String cmd = parseCommand(s);
+        int index = 0;
+        switch (cmd) {
+            case "deadline":
+                index = s.indexOf("/");
+                return s.substring(9, index - 1).trim();
```

Comment:
Perhaps could consider validating the input before parsing the description.

Any description that is taken in with less than 9 characters will throw StringIndexOutOfBounds exception and break the program, including but not limited to:

deadline /by sdf 
deadline /sdf /by sdf
deadline sdf /by
deadline /by